dart/test/protocol/rate_limiter_test.dart

```diff
@@ -0,0 +1,177 @@
+import 'package:test/test.dart';
+
+import 'package:sentry/src/transport/rate_limiter.dart';
+import 'package:sentry/src/protocol/sentry_event.dart';
+import 'package:sentry/src/sentry_envelope.dart';
+import 'package:sentry/src/sentry_envelope_header.dart';
+import 'package:sentry/src/sentry_envelope_item.dart';
+
+void main() {
+  var fixture = Fixture();
+
+  setUp(() {
+    fixture = Fixture();
+  });
+
+  test('uses X-Sentry-Rate-Limit and allows sending if time has passed', () {
+    final rateLimiter = fixture.getSUT();
+    fixture.dateTimeToReturn = 0;
+
+    final eventItem = SentryEnvelopeItem.fromEvent(SentryEvent());
+    final envelope =
+        SentryEnvelope(SentryEnvelopeHeader.newEventId(), [eventItem]);
+
+    rateLimiter.updateRetryAfterLimits(
+        '50:transaction:key, 1:default;error;security:organization', null, 1);
+
+    fixture.dateTimeToReturn = 1001;
+
+    final result = rateLimiter.filter(envelope);
+    expect(result, isNotNull);
+    expect(result!.items.length, 1);
+  });
+
+  test(
+      'parse X-Sentry-Rate-Limit and set its values and retry after should be true',
+      () {
+    final rateLimiter = fixture.getSUT();
+    fixture.dateTimeToReturn = 0;
+    final eventItem = SentryEnvelopeItem.fromEvent(SentryEvent());
+
+    // TODO Add another envelope item with different type and update rate limit header
+
+    // final transaction = SentryTransaction(
+    //     SentryTracer(TransactionContext('name', 'op'), mock()));
+    // final transactionItem =
+    //     SentryEnvelopeItem.fromEvent(fixture.serializer, transaction);
+    final envelope = SentryEnvelope(
+        SentryEnvelopeHeader.newEventId(), [eventItem /*, transactionItem*/]);
+
+    rateLimiter.updateRetryAfterLimits(
+        '50:transaction:key, 2700:default;error;security:organization',
+        null,
+        1);
+
+    final result = rateLimiter.filter(envelope);
+    expect(result, isNull);
+  });
+
+  test(
+      'parse X-Sentry-Rate-Limit and set its values and retry after should be false',
+      () {
+    final rateLimiter = fixture.getSUT();
+    fixture.dateTimeToReturn = 0;
+    final eventItem = SentryEnvelopeItem.fromEvent(SentryEvent());
+
+    // TODO Add another envelope item with different type and update rate limit header
```

Comment:
Is this still a valid TODO?

Reply:
This is for a followup, wehn we do support more envelope types like attachments. We only have event items for now.

Reply:
I'd prefer to remove such TODOs and commented code